tests/test_builders/test_build.py

```diff
@@ -68,7 +69,7 @@ def test_root_doc_not_found(tmp_path, make_app):
 
 
 @pytest.mark.sphinx(buildername='text', testroot='circular')
-def test_circular_toctree(app, status, warning):
+def test_circular_toctree(app, warning: str):
     app.build(force_all=True)
     warnings = warning.getvalue()
```

Comment:
`warning` is StringIO

Reply:
Yep, I'm aware. Just playing around at this stage to see whether the juice is worth the squeeze for this PR